### PR TITLE
Implement fixes for VTK 9.2.0rc

### PR DESCRIPTION
**Pull request by akaszynski**
Adds fixes for vtk 9.2 release candidate.

Regressions (VTK side):
- Opened https://gitlab.kitware.com/vtk/vtk/-/issues/18582 due to apparent regression in HDF reader. 

Other changes:
- Split up chart tests to aid in debugging.
- Change ``GetError`` to ``GetThresholdError`` for image comparison.
- Add note regarding flipping of cubemap direction between VTK 9.1.0 and 9.2.0.
- Also adds Python 3.10 to testing workflow.

### CODE REVIEW

File: tests/test_charts.py [hunk 1]
```diff
@@ -290,33 +310,37 @@ def test_axis(chart_2d):
...
     )
     assert axis.GetNotation() == charts.Axis.SCIENTIFIC_NOTATION
     assert axis.GetPrecision() == 4
-    axis.tick_locations = None
-    axis.tick_labels = None
-    chart_2d.show()
-    assert np.allclose(axis.tick_locations, tlocs0)
-    assert np.allclose(axis.GetTickPositions(), tlocs0)
-    assert tuple(axis.tick_labels) == tuple(tlabels0)
-    assert vtk_array_to_tuple(axis.GetTickLabels()) == tuple(tlabels0)
```
Comment (tkoyama010) [inline tests/test_charts.py:305]: Could you explain what makes this removed?

Comment (akaszynski) [reply tests/test_charts.py:305]: This will have to be resolved in #2868, but fundamentally I disagree with this test and I think that it only indicates that the chart plotting was already broken within VTK. Here, resetting the tick locations should set them to their defaults, but instead it resets it to their prior set locations. In VTK 9.1 this works (probably a bug), and in VTK 9.2 it resets them to their defaults (desired behavior).

So, in short, this portion of the test was removed because it is testing for invalid VTK 9.1 behavior.

Comment (tkoyama010) [reply tests/test_charts.py:305]: I got it. Thanks.

**Review by tkoyama010 — APPROVED** [review]
LGTM

File: pyvista/utilities/helpers.py [hunk 2]
```diff
@@ -1447,6 +1447,11 @@ def cubemap(path='', prefix='', ext='.jpg'):
     pyvista.Texture
         Texture with cubemap.
 
+    Notes
+    -----
+    Cubemap will appear flipped relative to the XY plane between VTK v9.1 and
+    VTK v9.2.
```
Comment (adeak) [inline pyvista/utilities/helpers.py:1453]: Yikes... We should remember to also note this in https://github.com/pyvista/pyvista/pull/2729. Do you know if this was intentional? If it's a bug perhaps it won't make it into 9.2.0, or will be reverted by 9.2.1...

File: tests/test_regression.py [hunk 3]
```diff
@@ -27,6 +27,11 @@ def test_compare_images_two_plotters_same(sphere, tmpdir):
     with pytest.raises(TypeError):
         pv.compare_images(im1, pl1.ren_win)
 
+    # test that this fails when the plotter is closed
+    pl1.close()
+    with pytest.raises(AttributeError, match='already been closed'):
```
Comment (adeak) [inline tests/test_regression.py:32]: Change this is the other end is changed
```suggestion
    with pytest.raises(RuntimeError, match='already been closed'):
```

File: tests/utilities/test_reader.py [hunk 4]
```diff
@@ -618,9 +618,9 @@ def test_openfoam_case_type():
         reader.case_type = 'wrong_value'
 
 
-@pytest.mark.skipif(pyvista.vtk_version_info < (9, 1), reason="Requires VTK v9.1.0 or newer")
+@pytest.mark.skipif(pyvista.vtk_version_info != (9, 1), reason="Requires exactly VTK v9.1.0")
```
[the remark this answers is not in the document]
Comment (akaszynski) [reply tests/utilities/test_reader.py:621]: At the moment, this is being investigated in [vtk #18582](https://gitlab.kitware.com/vtk/vtk/-/issues/18582). I'm assuming there's a regression on their end and as this is only a release candidate, there's probably some work to be done to fix this.

I'm going to keep this in to ensure testing works for Python 3.10, but [vtk #18582](https://gitlab.kitware.com/vtk/vtk/-/issues/18582) is resolved (or a workaround presented) we can implement the fix here.

[the remark this answers is not in the document]
Comment (akaszynski) [reply tests/utilities/test_reader.py:621]: Looks like there's a solution:
https://gitlab.kitware.com/vtk/vtk/-/issues/18582#note_1208581

File: pyvista/utilities/helpers.py (hunk 2 above)
Comment (banesullivan) [reply pyvista/utilities/helpers.py:1453]: Oof. This seems like a bug to me.

File: tests/utilities/test_reader.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (adeak) [reply tests/utilities/test_reader.py:621]: So the specification for HDF files has changed, and we can patch the existing example HDF file? Or am I misreading that thread?

[the remark this answers is not in the document]
Comment (akaszynski) [reply tests/utilities/test_reader.py:621]: I think we need to explicitly specify a "type" to the filter. Proof is in the pudding, so let me see if I can add it in.

File: pyvista/utilities/helpers.py (hunk 2 above)
Comment (akaszynski) [reply pyvista/utilities/helpers.py:1453]: > Oof. This seems like a bug to me.

@banesullivan, do you want to raise an issue or would you like me to?

File: pyvista/utilities/regression.py [hunk 5]
```diff
@@ -116,6 +116,10 @@ def to_img(img):
             if img._first_time:  # must be rendered first else segfault
                 img._on_first_render_request()
                 img.render()
+            if not hasattr(img, 'ren_win'):
+                raise AttributeError(
```
[the remark this answers is not in the document]
Comment (akaszynski) [reply pyvista/utilities/regression.py:120]: I'm going to accept this change, but then follow this up in a follow-up PR.

File: .github/workflows/testing-and-deployment.yml [hunk 6]
```diff
@@ -25,7 +25,7 @@ jobs:
     name: Mac OS Unit Testing
     strategy:
       matrix:
-        python-version: ['3.7', '3.8']
+        python-version: ['3.7', '3.10']
```
[the remark this answers is not in the document]
Comment (MatthewFlamm) [reply .github/workflows/testing-and-deployment.yml:28]: I'd prefer the structure in #2837 or similar, with the release candidate specifically defined outside the matrix.  Otherwise, it isn't going to be obvious to contributors what the subtleties are in the tests.  But it could happen here rather than #2837.

File: tests/utilities/test_reader.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (akaszynski) [reply tests/utilities/test_reader.py]: Turns out the format of the file itself changed...

I had build VTK from source and upload the new example file and add a version error for the old file in c137fb8.

**Comment by akaszynski** [comment]
@banesullivan, could you review so we could have this merged tomorrow? I'm hoping for a v0.35.1 (not v0.35.0 because I screwed that release up) by the end of the week.

**Review by banesullivan — APPROVED** [review]
LGTM
